src/lib/pubkey/pk_ops.cpp

```diff
@@ -45,7 +46,13 @@ secure_vector<uint8_t> PK_Ops::Decryption_with_EME::decrypt(uint8_t& valid_mask,
                                                             const uint8_t ciphertext[],
                                                             size_t ciphertext_len) {
    const secure_vector<uint8_t> raw = raw_decrypt(ciphertext, ciphertext_len);
-   return m_eme->unpad(valid_mask, raw.data(), raw.size());
+
+   secure_vector<uint8_t> ctext(raw.size());
+   auto len = m_eme->unpad(ctext, raw);
+
+   valid_mask = CT::Mask<uint8_t>::from_choice(len.has_value()).if_set_return(0xFF);
+   ctext.resize(len.value_or(0));
```

Comment:
That strikes me as dangerous!
If the `CT::Option` is set, this is probably not a (de)allocation, because the size of the initial allocation was -- more or less -- correct). If the `CT::Option` is not set, this might deallocate the vector's memory (as the `resize(0)` suggests).

Couldn't this turn into a padding oracle?

Reply:
In the long run, it would probably be better to not return a vector but let the user provide enough memory in an out-param and just return the number of bytes written, I guess.

Reply:
Wait `resize(0)` can cause a free? :(

Reply:
“Vector capacity is never reduced when resizing to smaller size because that would invalidate all iterators” - https://en.cppreference.com/w/cpp/container/vector/resize

Reply:
(And we’re always shrinking, never growing - indeed a reallocation would cause an oracle)

Reply:
Mhh, okay. That sounds authoritative enough to rely on. Perhaps leave a comment about this, though. And/or a `BOTAN_DEBUG_ASSERT(raw.size() >= len.value_or(0))` to really make this clear. 😅